tests/kubevirt_api_lifecycle_automation_test.go

```diff
@@ -0,0 +1,184 @@
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+		return
+	}
+}
+
+func beReady() gomegatypes.GomegaMatcher {
+	return gstruct.PointTo(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+		"Status": gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+			"Ready": BeTrue(),
+		}),
+	}))
+}
+
+func HaveMachineType(machineType string) gomegatypes.GomegaMatcher {
+	return gstruct.PointTo(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+		"Spec": gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+			"Template": gstruct.PointTo(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+				"Spec": gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+					"Domain": gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+						"Machine": gstruct.PointTo(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+							"Type": Equal(machineType),
+						})),
+					}),
+				}),
+			})),
+		}),
+	}))
+}
+
+// getAllSchedulableNodes returns list of Nodes which are "KubeVirt" schedulable.
+func getAllSchedulableNodes(virtClient *kubernetes.Clientset) *k8sv1.NodeList {
+	nodeList, err := virtClient.CoreV1().Nodes().List(context.Background(), v1.ListOptions{
+		LabelSelector: virt1.NodeSchedulable + "=" + "true",
+	})
+	Expect(err).ToNot(HaveOccurred(), "Should list compute nodeList")
+	return nodeList
+}
+
+// getSupportedMachineTypes returns list of supported machine types of a node
+func getSupportedMachineTypes(f *framework.Framework, targetNodeName string, kubevirtNS string) ([]libvirtxml.CapsGuestMachine, error) {
+	var machines []libvirtxml.CapsGuestMachine
+	var capabilities *libvirtxml.Caps
+	cmd := []string{"bash", "-c", fmt.Sprintf("cat /var/lib/kubevirt-node-labeller/capabilities.xml")}
+	stdout, stderr, err := libnode.ExecuteCommandOnNodeThroughVirtHandler(f, targetNodeName, kubevirtNS, cmd)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(stderr).To(BeEmpty())
+	Expect(stdout).ToNot(BeEmpty())
+	err = xml.Unmarshal([]byte(stdout), &capabilities)
+	if err != nil {
+		return nil, err
+	}
+	for _, guest := range capabilities.Guests {
+		machines = append(machines, guest.Arch.Machines...)
+	}
+	return machines, nil
+}
+
+// getDefaultMachineType the default machinetype for a node
+func getDefaultMachineType(f *framework.Framework, targetNodeName string) string {
+	c, err := f.GetVirtClient()
+	Expect(err).ToNot(HaveOccurred())
+	vmi := libvmifact.NewGuestless(
+		libvmi.WithNamespace(f.Namespace.Name),
+		libvmi.WithNodeSelectorFor(targetNodeName),
+	)
+	vm := libvmi.NewVirtualMachine(
+		vmi,
+		libvmi.WithRunStrategy(virt1.RunStrategyAlways),
+	)
+	vm, err = c.KubevirtV1().VirtualMachines(f.Namespace.Name).Create(context.Background(), vm, v1.CreateOptions{})
+	Expect(err).ToNot(HaveOccurred())
+	err = c.KubevirtV1().VirtualMachines(f.Namespace.Name).Delete(context.Background(), vm.Name, v1.DeleteOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	return vm.Spec.Template.Spec.Domain.Machine.Type
```

Comment:
@Barakmor1 You can use dry-run here. Mutating admitters should be applied in dry run.

Reply:
Yes, thanks for the great advice :)